crates/engine/tree/src/tree/mod.rs

```diff
@@ -2336,28 +2342,34 @@ where
         }
 
         let state_provider = provider_builder.build()?;
+
         // We only run the parallel state root if we are currently persisting blocks that are all
         // ancestors of the one we are executing. If we're committing ancestor blocks, then: any
         // trie updates being committed are a subset of the in-memory trie updates collected before
         // fetching reverts. So any diff in reverts (pre vs post commit) is already covered by the
         // in-memory trie updates we collect in `compute_state_root_parallel`.
         //
         // See https://github.com/paradigmxyz/reth/issues/12688 for more details
-        let is_descendant_of_persisting_blocks =
+        let descendant_of_persisting_blocks =
             self.is_descendant_of_persisting_blocks(block.header());
+        let run_parallel_state_root = descendant_of_persisting_blocks.is_not_persisting() ||
+            descendant_of_persisting_blocks.is_descendant();
```

Comment:
this isn't super straight forward imo

we have to do some mental gymnastic to find out which variant this is actually checks

can we make this a fn instead that matches!